pkg/client/client.go

```diff
@@ -121,6 +123,15 @@ func (c *client) Delete(ctx context.Context, obj runtime.Object, opts ...DeleteO
 	return c.typedClient.Delete(ctx, obj, opts...)
 }
 
+// Patch implements client.Client
+func (c *client) Patch(ctx context.Context, pt types.PatchType, data []byte, obj runtime.Object, subresources ...string) error {
```

Comment:
I think it would be better to make this a splat of PatchOptions like we have for ListOptions and DeleteOptions to allow for future expansion. We can then create an option method for subresource.

Reply:
our past discussed subresource plan has been a `.Subresource` method, but I'm open for debate on that.  Otherwise agree that using a splat for subresource isn't the right way to go

Reply:
I removed the  `subresources` splat and exchanged it for `PatchOptionFunc`. However, since there was no final clarity yet how subresources are handled, the `PatchOptions` are empty for now.
When we finally agree on how to do subresource handling in the future, we can incorporate it without changing the API.